lightning/src/chain/keysinterface.rs

```diff
@@ -1236,14 +1231,16 @@ impl KeysManager {
 
 impl EntropySource for KeysManager {
 	fn get_secure_random_bytes(&self) -> [u8; 32] {
-		let mut sha = self.rand_bytes_unique_start.clone();
+		let index = self.rand_bytes_index.fetch_add(1, Ordering::AcqRel);
 
-		let child_ix = self.rand_bytes_child_index.fetch_add(1, Ordering::AcqRel);
-		let child_privkey = self.rand_bytes_master_key.ckd_priv(&self.secp_ctx, ChildNumber::from_hardened_idx(child_ix as u32).expect("key space exhausted")).expect("Your RNG is busted");
-		sha.input(&child_privkey.private_key[..]);
+		let secs_plus_index = self.starting_time_secs + index as u64;
+		let mut nonce = Vec::new();
+		nonce.append(&mut secs_plus_index.to_be_bytes().to_vec());
```

Comment:
Adding an iterating index to `starting_time_secs` feels like it has some potential for generating duplicate nonces. 

Note that our docs currently state "`starting_time` isn’t strictly required to actually be a time, but it must absolutely, without a doubt, be unique to this instance. ie if you start multiple times with the same `seed`, `starting_time` must be unique to each run", which users might interpret as "I can leave the `starting_time_nanos` field 0, as long as I ensure the first part of it is unique". This unlikely scenario however might result in duplicate nonces, which should under no circumstances ever happen under the same seed. 

That said, I believe this would easily be resolved if we kept the `ChaCha20` state around, see below comment.

Reply:
It makes complete sense; thank you for pointing it out. Fixed in [d3f14e7](https://github.com/lightningdevkit/rust-lightning/pull/1974/commits/d3f14e7e2951892a70f72a4f2234416e1a10cb1f).